### PR TITLE
feat: add OpenTelemetry Tracing support as a preview feature

### PR DESCRIPTION
**Pull request by cojenco**
This adds OpenTelemetry Tracing support as a preview feature, and is subject to change before GA launch.  

To enable OpenTelemetry tracing in the python storage client, please refer to the readme for instructions and samples. Feel free to submit bugs and general feedback by opening a github issue.

### CODE REVIEW

File: google/cloud/storage/_opentelemetry_tracing.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
+"""Manages OpenTelemetry tracing span creation and handling."""
```
Comment (bajajneha27) [inline google/cloud/storage/_opentelemetry_tracing.py:1]: Copyright and License missing.

**Review by bajajneha27 — COMMENTED** [review]
LGTM with a tiny comment.
I'd let someone else also take a look since I don't know much about it.

File: google/cloud/storage/_opentelemetry_tracing.py (hunk 1 above)
Comment (cojenco) [reply google/cloud/storage/_opentelemetry_tracing.py]: Thanks Neha! I've added the copyright notice.

**Review by bajajneha27 — APPROVED** [review]

File: README.rst [hunk 2]
```diff
@@ -115,6 +115,61 @@ Windows
     .\<your-env>\Scripts\activate
     pip install google-cloud-storage
 
+
+Tracing With OpenTelemetry
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This is a PREVIEW FEATURE: Coverage and functionality are still in development and subject to change.
+
+This library uses `OpenTelemetry`_ to generate traces on calls to Google Cloud Storage.
```
Comment (andrewsg) [inline README.rst:124]: Let's try "This library can be configured to use..." to make sure it's clear that it's optional for people not familiar with Otel.

File: google/cloud/storage/_opentelemetry_tracing.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+_default_attributes = {
+    "rpc.service": "CloudStorage",
+    "rpc.system": "http",
+    "user_agent.original": f"gcloud-python/{__version__}",
+}
+
+
+@contextmanager
+def create_trace_span(
+    name, attributes=None, client=None, api_request=None, retry=None, **kwargs
```
Comment (andrewsg) [inline google/cloud/storage/_opentelemetry_tracing.py:60]: kwargs doesn't seem to be used here, what's its purpose?

**Review by andrewsg — CHANGES_REQUESTED** [review]

File: google/cloud/storage/_opentelemetry_tracing.py (hunk 3 above)
Comment (cojenco) [reply google/cloud/storage/_opentelemetry_tracing.py]: Removed for now; Will need it when we inject more span attributes in phase 2

File: README.rst (hunk 2 above)
Comment (cojenco) [reply README.rst]: Good idea, updated!

**Review by andrewsg — APPROVED** [review]

**Pull request merged**